src/api/apps/articles/test/model/save.test.coffee

```diff
@@ -85,6 +85,99 @@ describe 'Save', ->
         slugs: ['molly-clockwork']
       })
 
+    it 'does not append a date to the slug if it exists in the slugs array for that article already', (done) ->
+      Save.sanitizeAndSave( =>
+        Save.generateSlugs {
+          thumbnail_title: 'Clockwork'
+          published: true
+          slugs: ['molly-clockwork']
+          author: name: 'Molly'
+          published_at: '2017-07-26T17:37:03.065Z'
+        }, (err, article) =>
+          article.slugs.length.should.equal 1
+          article.slugs[0].should.equal 'molly-clockwork'
+          done()
+      )(null, {
+        slugs: ['molly-clockwork']
+      })
+
+    it 'appends the current date to the slug if the slug exists in the database already but the date is invalid', (done) ->
+      todays_date = moment().format('MM-DD-YY')
+      Save.sanitizeAndSave( =>
+        Save.generateSlugs {
+          thumbnail_title: 'Clockwork'
+          published: true
+          author: name: 'Molly'
+          published_at: '2017-07-dsdfdf26T17:37:03.065Zsdfdf'
+        }, (err, article) =>
+          article.slugs[0].should.equal 'molly-clockwork' + '-' + todays_date
+          done()
+      )(null, {
+        slugs: ['molly-clockwork']
+      })
+
+    it 'appends the current date to the slug if the slug exists in the database already but there is no publish date', (done) ->
+      todays_date = moment().format('MM-DD-YY')
+      Save.sanitizeAndSave( =>
+        Save.generateSlugs {
+          thumbnail_title: 'Clockwork'
+          published: true
+          author: name: 'Molly'
+        }, (err, article) =>
+          article.slugs[0].should.equal 'molly-clockwork' + '-' + todays_date
```

Comment:
My personal preference is for assertions like these to use explicit values rather than variables and interpolation. E.g., `...should.equal 'molly-clockwork-07-01-18'`. (This protects against the test passing while asserting against empty or buggy values.)

Not a big deal though.

Reply:
Can you think of a way to do this when testing that it appends today's date to the slug?